go-controller/pkg/node/gateway_localnet.go

```diff
@@ -29,6 +29,62 @@ const (
 	localnetGatewayExternalIDTable = "6"
 )
 
+func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net.IP, gwIntf string, nodeAnnotator kube.Annotator, recorder record.EventRecorder, cfg *managementPortConfig) (*gateway, error) {
+	gw := &gateway{}
+	var gatewayIfAddrs []*net.IPNet
+	for _, hostSubnet := range hostSubnets {
+		// local gateway mode uses mp0 as default path for all ingress traffic into OVN
+		var nextHop *net.IPNet
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			nextHop = cfg.ipv6.ifAddr
+		} else {
+			nextHop = cfg.ipv4.ifAddr
+		}
+		gatewayIfAddrs = append(gatewayIfAddrs, nextHop)
+
+		// add iptables masquerading for mp0 to exit the host for egress
+		cidr := nextHop.IP.Mask(nextHop.Mask)
+		cidrNet := &net.IPNet{IP: cidr, Mask: nextHop.Mask}
+		err := initLocalGatewayNATRules(util.K8sMgmtIntfName, cidrNet)
+		if err != nil {
+			return nil, fmt.Errorf("failed to add local NAT rules for: %s, err: %v", util.K8sMgmtIntfName, err)
+		}
+	}
+
+	bridgeName, uplinkName, _, err := gatewayInitInternal(
+		nodeName, gwIntf, hostSubnets, gwNextHops, nodeAnnotator)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.Gateway.NodeportEnable {
+		localAddrSet, err := getLocalAddrs()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := initLocalGatewayIPTables(); err != nil {
+			return nil, err
+		}
+		if err := initRoutingRules(); err != nil {
+			return nil, err
+		}
+		gw.localPortWatcher = newLocalPortWatcher(gatewayIfAddrs, recorder, localAddrSet)
+	}
+
+	gw.initFunc = func() error {
+		if !config.Gateway.NodeportEnable {
```

Comment:
i think this was a bug in the old code. These rules need to be there whether or not nodePort is enabled for local gateway mode. Really we should also enable them regardless for shared gw, but those flows are kind of up in the air right now and we'll be reworking them so can hold off on shared and just fix this here in local.

Reply:
Fixed in local gw.